src/popup.tsx

```diff
@@ -159,14 +193,60 @@ const Popup = () => {
                 setTypes(newTypes);
               }}
             />
-
+            <div
+              ref={(node) => {
+                colorsRef.current[idx] = node;
+              }}
+              className={`bg-${colors[idx]}-500 w-4 h-4`}
+              style={{
+                backgroundColor: colors[idx],
+                flexShrink: 0,
+              }}
+            ></div>
+            <svg
+              t="1702386306969"
+              class="icon"
+              viewBox="0 0 1024 1024"
+              version="1.1"
+              xmlns="http://www.w3.org/2000/svg"
+              p-id="1448"
+              width="16"
+              height="16"
+              onClick={(e) => {
+                let nextIdx =
+                  (DEFAULT_COLOR.indexOf(colors[idx]) + 1) %
+                  DEFAULT_COLOR.length;
+                const newColors = [...colors];
+                newColors[idx] = DEFAULT_COLOR[nextIdx];
+                colorsRef.current[idx].animate(
```

Comment:
it seems that you are using refs only for transition for colors? You can just add `transition-colors` in tailwind to do so. If there is any other reason using refs pls let me know~

Reply:
Yes, you're right. I was planning to utilize refs to create a translation animation with transform. The sole advantage of using refs might be the convenience it offers when altering the animation later on. Should I delete it...